src/StructureFactors/StructureFactors.jl

```diff
@@ -53,35 +114,85 @@ function StructureFactor(sys::System{N}; Δt, nω, measperiod,
                             process_trajectory = :none) where {N}
 
     # Set up correlation functions (which matrix elements αβ to save from 𝒮^{αβ})
-    default_observables = false
-    default_correlations = false
     if isnothing(observables)
-        observables = zeros(ComplexF64, 0, 0, 3)  # observables are empty in this case
-        default_observables = true
+        # Default observables are spin x,y,z
+        # projections (SU(N) mode) or components (dipole mode)
+        observable_ixs = Dict(:Sx => 1,:Sy => 2,:Sz => 3)
+        if N == 0
+            dipole_component(i) = FunctionMap{Float64}(s -> s[i],1,3)
+            observables = dipole_component.([1,2,3])
+        else
+            # SQTODO: Make this use the more optimized expected_spin function
+            # Doing this will also, by necessity, allow users to make the same
+            # type of optimization for their vector-valued observables.
+            observables = LinearMap{ComplexF64}.(spin_matrices(N))
```

Comment:
What is the motivation for using this instead of a dense `Hermitian{ComplexF64}`? If all else is equal, I'd prefer the explicit option. One could imagine that `LinearMap` could be useful for optimization if we have a very fast matrix-vector product available, but will that be the case in our intended applications?

Reply:
The main reason is that this (should) compile to the same thing as it did before this change specifically for dipole mode. A linear map `s -> s[i]` is better than multiplying by a row vector IMO. It's also nicer for advanced users to specify a linear map, especially in sun mode

Reply:
Do you have examples in mind of linear maps that are not explicit matrices?

Reply:
BTW, one place this might show up is in LSWT where we can in principle construct unitaries `exp(I theta S)` as a polynomial of sparse spin operators S. But maybe that type of thing never appears for observables.